addon/components/polaris-choice-list.js

```diff
@@ -99,11 +117,27 @@ export default Component.extend({
    *
    * @property onChange
    * @public
-   * @type {function}
+   * @type {Function}
    * @default noop
    */
   onChange() {},
 
+  'data-test-choice-list': true,
```

Comment:
just realized the other day too, having and then passing in something like `data-test-*` into a tagless component will result in a warning, so instead we should use a camelCased name
but, you're also not hooking this in with the first element in the template, so no value in having it if we're not using it...either hook it in or remove all together?